app/src/main/java/com/example/habitsmasher/PublicPrivateButtons.java

```diff
@@ -0,0 +1,131 @@
+package com.example.habitsmasher;
+
+import static android.content.ContentValues.TAG;
+
+import android.nfc.Tag;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.CheckBox;
+
+import com.google.android.material.tabs.TabLayout;
+
+public class PublicPrivateButtons {
+    /**
+     * This class handles the logic behind the public and private buttons, which are used to
+     * set a habit to either public or private.
+     *
+     * @author Cameron Matthew
+     * @version 1.0
+     */
+
+    private Button _publicButton;
+    private Button _privateButton;
+    private boolean _publicSelected;
+    private boolean _privateSelected;
+
+    /**
+     * This constructor is the default constructor, and is used in {@link com.example.habitsmasher.ui.dashboard.AddHabitDialog}.
+     * @param view The view that has the buttons.
+     */
+    public PublicPrivateButtons(View view){
+        // get the buttons
+        _publicButton = view.findViewById(R.id.public_button);
+        _privateButton = view.findViewById(R.id.private_button);
+
+        // set the public button as the default button.
+        _publicButton.performClick();
+        _publicButton.setClickable(false);
+        _publicSelected = true;
+        _privateSelected = false;
+        Log.d("BUTTONS", "Standard const., public="+_publicSelected+", private="+_privateSelected);
+        Log.d("BUTTONS", "Is button clickable? " + _publicButton.isClickable());
+
+    }
+
+    /**
+     * This constructor is used whenever a habit is being viewed after creation, such as in {@link com.example.habitsmasher.ui.dashboard.EditHabitFragment}.
+     * @param view The view that has the buttons.
+     * @param isPublic Whether the habit is public or not.
+     */
+    public PublicPrivateButtons(View view, boolean isPublic){
+        // get the buttons
+        _publicButton = view.findViewById(R.id.public_button);
+        _privateButton = view.findViewById(R.id.private_button);
+
+        Log.d(TAG, String.valueOf(isPublic));
+
+        // if its public, click the public button. if not, click the private button
+        if (isPublic){
+            Log.d(TAG, "Habit is public.");
+            _publicButton.performClick();
+            _publicButton.setClickable(false);
+            _publicSelected = true;
+            _privateSelected = false;
+        }
+        else{
+            Log.d(TAG, "Habit is private.");
+            _privateButton.performClick();
+            _privateButton.setClickable(false);
+            _publicSelected = false;
+            _privateSelected = true;
+        }
```

Comment:
Starting logic for the buttons. This is bugged, since the button is still clickable despite setting it as unclickable.

Reply:
Are these Log statements just for debugging? Also do you want to pair on how to fix this bug? I think it would probably be best not to introduce code with bugs into develop

Reply:
It's not a big deal if we can unclick the button tbh, as long as we are safeguarding the user from entering a habit **without anything selected**. 

Do we make this a requirement in the Habit validation process?

Reply:
I figured that if the buttons couldn't be unclicked, there wouldn't be a need to check whether either button is selected. I can go back and change everything to be uncheckable if everyone wants me to.